NiceIO.Tests/Compare.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using NUnit.Framework;
+
+namespace NiceIO.Tests
+{
+	[TestFixture]
+	public class Compare
+	{
+		[Test]
+		public void FullPathToSameFullPath()
+		{
+			var path1 = NPath.CurrentDirectory.Combine("myfile.txt");
+			var path2 = NPath.CurrentDirectory.Combine("myfile.txt");
+
+			Assert.That(path1.CompareTo(path2), Is.EqualTo(0));
+		}
+
+		[Test]
+		public void FullPathToNull()
+		{
+			var path1 = NPath.CurrentDirectory.Combine("myfile.txt");
+
+			Assert.That(path1.CompareTo(null), Is.EqualTo(-1));
+		}
+
+		[Test]
+		[Platform(Include = "Win")]
+		public void FullPathToSameFullPath_DifferentSlashes()
+		{
+			var path1 = NPath.CurrentDirectory.Combine("myfile.txt");
+			var path2 = new NPath(NPath.CurrentDirectory.Combine("myfile.txt").ToString().Replace('\\', '/'));
+
+			Assert.That(path1.CompareTo(path2), Is.EqualTo(0));
+		}
+
+		[Test]
+		public void FullPathToRelativePathOfSameLocation()
+		{
+			var path1 = NPath.CurrentDirectory.Combine("myfile.txt");
+			var path2 = new NPath("myfile.txt");
+
+			Assert.That(path1.CompareTo(path2), Is.EqualTo(0));
```

Comment:
I guess that was just the behavior that seemed sensible at the time.  But now that you point it out, I can see where a more literal value comparison approach would be appropriate.  Then require that the user make everything absolute before hand if they want this behavior. 

I will open a new PR to address this.